pkg/query-service/app/logs/v3/query_builder.go

```diff
@@ -376,33 +366,37 @@ func reduceQuery(query string, reduceTo v3.ReduceToOperator, aggregateOperator v
 	return query, nil
 }
 
-func addLimitToQuery(query string, limit uint64, panelType v3.PanelType) string {
-	if limit == 0 {
-		limit = 100
-	}
-	if panelType == v3.PanelTypeList {
-		return fmt.Sprintf("%s LIMIT %d", query, limit)
-	}
-	return query
+func addLimitToQuery(query string, limit uint64) string {
+	return fmt.Sprintf("%s LIMIT %d", query, limit)
 }
 
 func addOffsetToQuery(query string, offset uint64) string {
 	return fmt.Sprintf("%s OFFSET %d", query, offset)
 }
 
-func PrepareLogsQuery(start, end int64, queryType v3.QueryType, panelType v3.PanelType, mq *v3.BuilderQuery, fields map[string]v3.AttributeKey) (string, error) {
-	query, err := buildLogsQuery(start, end, mq.StepInterval, mq, fields)
+func PrepareLogsQuery(start, end int64, queryType v3.QueryType, panelType v3.PanelType, mq *v3.BuilderQuery) (string, error) {
+	query, err := buildLogsQuery(panelType, start, end, mq.StepInterval, mq)
 	if err != nil {
 		return "", err
 	}
 	if panelType == v3.PanelTypeValue {
 		query, err = reduceQuery(query, mq.ReduceTo, mq.AggregateOperator)
 	}
 
-	query = addLimitToQuery(query, mq.Limit, panelType)
-
-	if mq.Offset != 0 {
-		query = addOffsetToQuery(query, mq.Offset)
+	if panelType == v3.PanelTypeList || panelType == v3.PanelTypeTable {
```

Comment:
Read this https://signoz-team.slack.com/archives/C02B85R4R0A/p1685603948437199. You will have to update this for table view.

Reply:
So from the convesation it seems that the limit for table will be similar to timeseries and implementation has to be decided right ?

Let me remove paneltype from the above condition